File: php/L11/model/Prodotto.php
<?php

class Prodotto
{
	public function __construct(
		private $nome,
		private $categoria,
		private $prezzo,
		private $giacenza
	) {}

	public function __get($property)
	{
		return $this->$property;
	}

	public function __set($property, $value)
	{
		$this->$property = $value;
	}

	public function __toString(){
		return "Nome: $this->nome, Categoria: $this->categoria, Prezzo: $this->prezzo, Giacenza: $this->giacenza";
	}

}

File: Java/2025-2026/Prj_16B_VScode/magazzino/src/main/java/com/querio/entities/Prodotto.java
package com.querio.entities;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Data;


@Entity
@Table(name = "prodotti")
@Data
public class Prodotto {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int id;
	private String nome;
	private String categoria;
	private double prezzo;
	private int giacenza;

}
